lib/src/prayer_times.dart

```diff
@@ -108,178 +108,161 @@ class PrayerTimes {
 
   PrayerTimes._(this.coordinates, DateTime _date, this.calculationParameters,
       {this.utcOffset}) {
+    bool _valid(double value) => !(value.isInfinite || value.isNaN);
+
+    late double _value;
+
     final date = _date.toUtc();
     _dateComponents = DateComponents.from(date);
 
-    DateTime tempFajr;
-    DateTime tempSunrise;
-    DateTime tempDhuhr;
-    DateTime tempAsr;
-    DateTime tempMaghrib;
-    DateTime tempIsha;
+    DateTime? tempFajr;
+    late DateTime tempSunrise;
+    late DateTime tempDhuhr;
+    late DateTime tempAsr;
+    late DateTime tempMaghrib;
+    DateTime? tempIsha;
 
     final year = date.year;
     final dayOfYear = date.dayOfYear;
 
     final solarTime = SolarTime(date, coordinates);
 
     var timeComponents = TimeComponents.fromDouble(solarTime.transit);
```

Comment:
TimeComponents.fromDouble is throwing ArgumentError and that is not helpful to the library user, We need to handle those internal errors and throw meaningful exceptions.

Reply:
ArgumentError tells that value of fromDouble() should not be infinite or NaN. This can only happen if the coordinated value is either infinite or NaN. The user can handle this easily.

Reply:
We should force coordinates validation to prevent this kind of behavior.

Reply:
we can force Coordinates to throw Error by keeping the value of `validate` in `Coordinate` true as default value. 
In either case, if validate is true.. Coordinate will throw an error else if validate is false, fromDouble will throw an error.